### PR TITLE
Ability to initialize simulations from vtk grain ID data

### PR DESCRIPTION
**Pull request by MattRolchigo**
Removed the deprecated ability to read the old substrate grain ID data format and replaced it with the ability to read grain ID data from a vtk file

### CODE REVIEW

**Comment by streeve** [comment]
Looks like this should check off part of #22 (maybe more of those are already done too?)

File: src/CAparsefiles.cpp [hunk 1]
```diff
@@ -170,3 +170,24 @@ std::size_t checkForHeaderValues(std::string header_line) {
...
+std::vector<std::string> readVTKTuple(std::ifstream &input_data_stream) {
+    std::vector<std::string> read_values(3);
+    std::string read_line;
+    getline(input_data_stream, read_line);
+    std::size_t first_separator = read_line.find(' ');
+    std::size_t second_separator = read_line.find(' ', first_separator + 1);
+    std::size_t third_separator = read_line.find(' ', second_separator + 1);
+    read_values[0] = read_line.substr(first_separator, second_separator - first_separator);
+    read_values[1] = read_line.substr(second_separator, third_separator - second_separator);
+    read_values[2] = read_line.substr(third_separator);
```
Comment (streeve) [inline src/CAparsefiles.cpp:191]:
```suggestion
    char separator = ' ';
    std::size_t line_size = read_line.size();
    for (std::size_t n = 0; n < line_size - 1; n++) {
        std::size_t pos = read_line.find(separator);
        read_values[n] = read_line.substr(0, pos);
        read_line = read_line.substr(pos + 1, std::string::npos);
    }
```
Trying to make it more readable and reusable

Comment (MattRolchigo) [reply src/CAparsefiles.cpp:191]: Now that I think about it, we already have `splitString` that can do this exact thing...

File: src/CAparsefiles.cpp [hunk 2]
```diff
@@ -82,6 +82,35 @@ void splitString(const std::string line, std::vector<std::string> &parsed_line,
...
+    getline(input_data_stream, line);
+
+    // Make sure the right number of values are present on the line - one more than the number of separators
+    std::size_t actual_num_values = std::count(line.begin(), line.end(), separator) + 1;
+    if (expected_num_values != actual_num_values) {
+        std::string error = "Error: Expected " + std::to_string(expected_num_values) +
+                            " values while reading file; but " + std::to_string(actual_num_values) + " were found";
+        throw std::runtime_error(error);
+    }
+    // Separate the line into its components, now that the number of values has been checked
```
Comment (streeve) [inline src/CAparsefiles.cpp:100]: Can't you just call `splitString`?

Comment (MattRolchigo) [reply src/CAparsefiles.cpp:100]: You mean call `void splitString` from within `std::vector<std::string splitString`?

Comment (streeve) [reply src/CAparsefiles.cpp:100]: Yeah that's what I meant

**Review by streeve — APPROVED** [review]

**Pull request merged**